RealmSwift/Aliases.swift

```diff
@@ -17,7 +17,7 @@
 ////////////////////////////////////////////////////////////////////////////
 
 import Foundation
-import Realm
+import Realm.Swift
```

Comment:
Using an aliased type from a resilient module (which SPM builds some things as) without importing Realm now gives "'Object' aliases 'Realm.RealmSwiftObject' and cannot be used here because 'Realm' was not imported by this file; this is an error in Swift 6". `@_exported import Realm` re-exports way too much, so I moved RealmSwiftObject to a separate submodule which doesn't export too much. I'm not sure why it doesn't need `@_exported`. This might just be making things too complicated for the warning to work rather than fixing the problem.